cli/tests/environment-pull.bats

```diff
@@ -300,3 +326,50 @@ function make_incompatible() {
   run "$FLOX_BIN" list
   assert_success
 }
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=pull:unsupported-package
+# pulling an environment with a package that is not available for the current platform
+# should fail with an error
+@test "pull environment with package not available for the current platform fails" {
+  update_dummy_env "owner" "name"
+  add_incompatible_package "owner" "name"
+
+  run "$FLOX_BIN" pull --remote owner/name
+
+  assert_failure
+  assert_line --partial "'extra' is not available on the requested hostPlatform ('$NIX_SYSTEM')"
+}
+
+# bats test_tags=pull:unsupported-package:prompt-fail
+# pulling an environment with a package that is not available for the current platform
+# should prompt to ignore the error and pull the environment anyway.
+# When answering no, an error should be shown and the environment should not be pulled.
+@test "pull environment with package not available for the current platform prompts to abort or ignore -- aborts cleanly" {
+  update_dummy_env "owner" "name"
+  add_incompatible_package "owner" "name"
+
+  run -0 expect -d "$TESTS_DIR/pull/answerPrompt.exp" owner/name "$UNSUPPORTED_PACKAGE_PROMPT" no
+  assert_success
+  assert_line --partial "'extra' is not available on the requested hostPlatform ('$NIX_SYSTEM')"
+  assert_output --partial "$UNSUPPORTED_PACKAGE_PROMPT"
+  assert_line --partial "Did not pull the environment."
+}
+
+# bats test_tags=pull:unsupported-package:prompt-success
+# pulling an environment with a package that is not available for the current platform
+# should prompt to ignore the error and pull the environment anyway.
+# When answering yes, the environment should be pulled in a potentially broken state.
+@test "pull environment with package not available for the current platform prompts to abort or ignore -- ingores" {
+  update_dummy_env "owner" "name"
+  add_incompatible_package "owner" "name"
+
+  run -0 expect -d "$TESTS_DIR/pull/answerPrompt.exp" owner/name "$UNSUPPORTED_PACKAGE_PROMPT" yes
+  assert_success
+  assert_line --partial "'extra' is not available on the requested hostPlatform ('$NIX_SYSTEM')"
+  assert_output --partial "$UNSUPPORTED_PACKAGE_PROMPT"
+
+  run "$FLOX_BIN" list
+  assert_success
+}
```

Comment:
question blocking: is there test coverage for more than just one of the 3 cases you added? Since some of the errors are handled differently, I think we should make sure we cover them all. The error looks good for `gdb`, but I'm not sure about the other cases

Reply:
I'll add a test for _insecure_ packages which is a whole other eval failure group that we might want to catch explicitly anyway (not in this PR necessarily).

For the build error case i dont think there's anything in nixpkgs that is _not_ marked as broken (and will fail at eval) and does not build locally.
This will be more of an issue later on